### PR TITLE
Prevent closed claims with open evidence requests from bubbling up

### PR DESCRIPTION
**Pull request by kjduensing**
A previous commit introduced a bug where, if a claim was closed and still had open evidence requests, the claim would still be sorted as a claim that needs attention.

## Related issue(s)
department-of-veterans-affairs/va.gov-team#64880

### CODE REVIEW

File: src/applications/claims-status/tests/e2e/fixtures/mocks/mock-api/claims/claims-list.json [hunk 1]
```diff
@@ -88,6 +88,72 @@
         "updatedAt": "2023-03-30T18:19:29.857Z"
       }
     },
+    {
```
Comment (jerekshoe) [inline src/applications/claims-status/tests/e2e/fixtures/mocks/mock-api/claims/claims-list.json:91]: I don't think it needs to hold this PR up, but we should make sure that there are json files that correspond to the show returns for all 3 of these, i.e, closed-with-documents-needed.json, phase3-w-documents-needed.json, phase4-w-documents-needed.json

File: src/applications/claims-status/tests/e2e/fixtures/mocks/mock-api/claims/claims-list.json [hunk 2]
```diff
@@ -88,6 +88,72 @@
...
+      "id": "closed-with-documents-needed",
```
Comment (jerekshoe) [inline src/applications/claims-status/tests/e2e/fixtures/mocks/mock-api/claims/claims-list.json:92]: Can we change this to `closed-w-documents-needed`? I realize I'm nitpicking but would be nice to maintain the consistency

File: src/applications/claims-status/tests/utils/helpers.unit.spec.js [hunk 3]
```diff
@@ -44,10 +44,11 @@ describe('Disability benefits helpers: ', () => {
     const claims = [
       {
         claimId: 1,
-        type: 'claim',
+        type: 'evss_claims',
```
Comment (jerekshoe) [inline src/applications/claims-status/tests/utils/helpers.unit.spec.js:47]: EVSS doesn't have a claimPhaseDates object, that is a LH thing

Comment (jerekshoe) [reply src/applications/claims-status/tests/utils/helpers.unit.spec.js:47]: Unless this is intentionally like this

File: src/applications/claims-status/utils/helpers.js [hunk 4]
```diff
@@ -52,6 +65,22 @@ export function getUserPhase(phase) {
   return phase - 3;
 }
 
+function isInEvidenceGathering(claim) {
+  const allowedClaimTypes = ['evss_claims', 'claim'];
+  const isEvssClaim = claim.type.toLowerCase() === 'evss_claims';
```
Comment (jerekshoe) [inline src/applications/claims-status/utils/helpers.js:70]: I don't think we need to worry about lowercasing these, they should always be either either `evss_claims` or `claim`. For EVSS for example, the type field is set by vets-api

File: src/applications/claims-status/utils/helpers.js [hunk 5]
```diff
@@ -21,6 +21,19 @@ const phaseMap = {
   8: 'Complete',
 };
 
+const lighthouseStatuses = [
```
Comment (jerekshoe) [inline src/applications/claims-status/utils/helpers.js:24]: I'm wondering if it would be better to use the expanded statuses we get from LH instead of the condensed ones. Thoughts?

File: src/applications/claims-status/utils/helpers.js [hunk 6]
```diff
@@ -148,12 +177,14 @@ export function displayFileSize(size) {
...
 
 export function groupClaimsByDocsNeeded(list) {
-  const claimsWithOpenRequests = list.filter(
-    claim => claim.attributes.documentsNeeded,
-  );
+  const groupingPredicate = c => {
+    return c.attributes.documentsNeeded && isInEvidenceGathering(c);
+  };
+
+  const claimsWithOpenRequests = list.filter(claim => groupingPredicate(claim));
```
Comment (jerekshoe) [inline src/applications/claims-status/utils/helpers.js:184]: This is pedantic, but you can just do this:
`const claimsWithOpenRequests = list.filter(groupingPredicate);`

File: src/applications/claims-status/utils/helpers.js [hunk 7]
```diff
@@ -328,7 +359,9 @@ export function getCompletedDate(claim) {
 }
 
 export function getClaimType(claim) {
-  return claim?.attributes?.claimType || 'Disability Compensation';
+  return (
+    claim?.attributes?.claimType || 'disability compensation'
+  ).toLowerCase();
```
Comment (jerekshoe) [inline src/applications/claims-status/utils/helpers.js:364]: I don't think this change should be in here

File: src/applications/claims-status/utils/helpers.js [hunk 8]
```diff
@@ -235,7 +266,7 @@ export function getDocTypeDescription(docType) {
 }
 
 export const isPopulatedClaim = ({ claimDate, claimType, contentions }) =>
-  !!claimType && (contentions && !!contentions.length) && !!claimDate;
+  !!claimType && contentions && !!contentions.length && !!claimDate;
```
Comment (jerekshoe) [inline src/applications/claims-status/utils/helpers.js:269]: Is this change intentional?

File: src/applications/claims-status/tests/e2e/fixtures/mocks/mock-api/claims/claims-list.json (hunk 1 above)
Comment (kjduensing) [reply src/applications/claims-status/tests/e2e/fixtures/mocks/mock-api/claims/claims-list.json:91]: Noted, that'd a good point. I'll make a ticket for it.

File: src/applications/claims-status/utils/helpers.js [hunk 9]
```diff
@@ -1,11 +1,11 @@
 import merge from 'lodash/merge';
-import { format, isValid, parseISO } from 'date-fns';
 // import * as Sentry from '@sentry/browser';
 
-import environment from 'platform/utilities/environment';
+import environment from '@department-of-veterans-affairs/platform-utilities/environment';
 // import localStorage from 'platform/utilities/storage/localStorage';
 import { apiRequest } from 'platform/utilities/api';
 // import { fetchAndUpdateSessionExpiration as fetch } from 'platform/utilities/api';
+import { format, isValid, parseISO } from 'date-fns';
```
Comment (jerekshoe) [inline src/applications/claims-status/utils/helpers.js:8]: Can you move these imports back up to the top?. Other than that things looks good

**Review by jerekshoe — APPROVED** [review]
